src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -53105,6 +53090,66 @@ static void G(object? x, object? y)
                 Diagnostic(ErrorCode.ERR_NameNotInContext, "F").WithArguments("F").WithLocation(9, 36));
         }
 
+        [Fact]
+        [WorkItem(35010, "https://github.com/dotnet/roslyn/issues/35010")]
+        public void Tuple_Assignment_21()
+        {
+            var source =
+@"
+#pragma warning disable CS0219
+class C
+{
+    static void F(string? x, string? y)
+    {
+        (object a, long b) t = (x, 0)/*T:(string? x, int)*/; // 1
+        (object a, (long b, object c)) t2 = (x, (0, y)/*T:(int, string? y)*/)/*T:(string? x, (int, string? y))*/; // 2, 3
+        (object a, (long b, object c)) t3 = (x!, (0, y!)/*T:(int, string!)*/)/*T:(string!, (int, string!))*/;
+        (int b, string? c)? t4 = null;
+		(object? a, (long b, object? c)?) t5 = (x, t4)/*T:(string? x, (int b, string? c)? t4)*/; 
+
+    }
+}";
+            var comp = CreateCompilation(source, options: WithNonNullTypesTrue());
+            comp.VerifyDiagnostics(
+              // (7,32): warning CS8619: Nullability of reference types in value of type '(object? x, long)' doesn't match target type '(object a, long b)'.
+              //         (object a, long b) t = (x, 0)/*T:(string? x, int)*/; // 1
+              Diagnostic(ErrorCode.WRN_NullabilityMismatchInAssignment, "(x, 0)").WithArguments("(object? x, long)", "(object a, long b)").WithLocation(7, 32),
+              // (8,45): warning CS8619: Nullability of reference types in value of type '(object? x, (long b, object c))' doesn't match target type '(object a, (long b, object c))'.
+              //         (object a, (long b, object c)) t2 = (x, (0, y)/*T:(int, string? y)*/)/*T:(string? x, (int, string! y))*/; // 2, 3
+              Diagnostic(ErrorCode.WRN_NullabilityMismatchInAssignment, "(x, (0, y)/*T:(int, string? y)*/)").WithArguments("(object? x, (long b, object c))", "(object a, (long b, object c))").WithLocation(8, 45),
+              // (8,49): warning CS8619: Nullability of reference types in value of type '(long, object? y)' doesn't match target type '(long b, object c)'.
+              //         (object a, (long b, object c)) t2 = (x, (0, y)/*T:(int, string? y)*/)/*T:(string? x, (int, string! y))*/; // 2, 3
+              Diagnostic(ErrorCode.WRN_NullabilityMismatchInAssignment, "(0, y)").WithArguments("(long, object? y)", "(long b, object c)").WithLocation(8, 49)
+              );
+            comp.VerifyTypes();
+        }
+
+        [Fact]
+        [WorkItem(35010, "https://github.com/dotnet/roslyn/issues/35010")]
+        public void Tuple_Assignment_22()
+        {
+            var source =
+@"
+#pragma warning disable CS0219
+class C
+{
+    static void F(string? x)
+    {
+     //   C/*T:C!*/ a = x/*T:string?*/;
+        //(C, C) a = (new C(), new C());
```

Comment:
Should these lines be uncommented or removed?

Reply:
Mistakenly included, will remove, thanks :)

---
In reply to: [296374978](https://github.com/dotnet/roslyn/pull/36628#discussion_r296374978) [](ancestors = 296374978)